Framework/Core/src/WorkflowHelpers.cxx

```diff
@@ -732,10 +732,7 @@ void WorkflowHelpers::injectServiceDevices(WorkflowSpec& workflow, ConfigContext
     std::vector<InputSpec> ignored = unmatched;
     ignored.insert(ignored.end(), redirectedOutputsInputs.begin(), redirectedOutputsInputs.end());
     for (auto& ignoredInput : ignored) {
-      if (ignoredInput.lifetime == Lifetime::OutOfBand) {
```

Comment:
Actually, I am not sure this is correct. The inputs should have the lifetime of the matching output, so they should have already matched...